src/engine/Join.cpp

```diff
@@ -176,27 +178,34 @@ Result Join::computeResult(bool requestLaziness) {
 VariableToColumnMap Join::computeVariableToColumnMap() const {
   return makeVarToColMapForJoinOperation(
       _left->getVariableColumns(), _right->getVariableColumns(),
-      {{_leftJoinCol, _rightJoinCol}}, BinOpType::Join,
-      _left->getResultWidth());
+      {{_leftJoinCol, _rightJoinCol}}, BinOpType::Join, _left->getResultWidth(),
+      _keepJoinColumn);
 }
 
 // _____________________________________________________________________________
 size_t Join::getResultWidth() const {
-  size_t res = _left->getResultWidth() + _right->getResultWidth() - 1;
-  AD_CONTRACT_CHECK(res > 0);
+  size_t res = _left->getResultWidth() + _right->getResultWidth() - 1 -
+               static_cast<size_t>(!_keepJoinColumn);
+  AD_CONTRACT_CHECK(res > 0 || !_keepJoinColumn);
```

Comment:
I don't think this check has much value anymore. If one of the children returns 0 for `getResultWidth()` we have an underflow and the check is true. It would make more sense to assert that both children are at least 1 in width, or drop this check entirely because the constructor should already eliminate this case (unless the implementation is simply wrong)